File: content/develop/use-cases/semantic-cache/java-jedis/src/main/java/com/redis/semcache/LookupResult.java
package com.redis.semcache;

/**
 * Sealed result of a cache lookup. Pattern-matched in the demo
 * server to branch between the hit and miss paths; mirrors the
 * {@code CacheHit | CacheMiss} union the Python and Node ports
 * return.
 */
public sealed interface LookupResult permits CacheHit, CacheMiss {
}

File: content/develop/use-cases/semantic-cache/dotnet/LookupResult.cs
namespace SemanticCacheDemo;

/// <summary>
/// Result of a cache lookup. Either a <see cref="CacheHit"/> or a
/// <see cref="CacheMiss"/>; pattern-matched in the demo server to
/// branch between the hit and miss paths. Mirrors the
/// <c>CacheHit | CacheMiss</c> union the Python and Node ports
/// return.
/// </summary>
public abstract record LookupResult;
